tests/platform_tests/daemon/test_thermalctld.py

```diff
@@ -0,0 +1,208 @@
+    thermalctld_pooling_interval = 60
+    wait_until(thermalctld_pooling_interval, 6, 0, _collect_data)
+    return shared_scope.data_after_restart
+
+@pytest.fixture(scope='module')
+def data_before_restart(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    data = collect_data(duthost)
+    return data
+
+
+def test_pmon_thermalctld_running_status(duthosts, rand_one_dut_hostname, data_before_restart):
+    """
+    @summary: This test case is to check thermalctld status on dut
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    logger.info("{} daemon is {} with pid {}".format(daemon_name, daemon_status, daemon_pid))
+    pytest_assert(daemon_status == expected_running_status,
+                          "{} expected running status is {} but is {}".format(daemon_name, expected_running_status, daemon_status))
+    pytest_assert(daemon_pid != -1,
+                          "{} expected pid is a positive integer but is {}".format(daemon_name, daemon_pid))
+
+    pytest_assert(data_before_restart['keys'], "DB keys is not availale on daemon running")
+    pytest_assert(data_before_restart['data'], "DB data is not availale on daemon running")
+
+
+def test_pmon_thermalctld_stop_and_start_status(check_daemon_status, duthosts, rand_one_dut_hostname, data_before_restart):
+    """
+    @summary: This test case is to check the thermalctld stopped and restarted status 
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    pre_daemon_status, pre_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    logger.info("{} daemon is {} with pid {}".format(daemon_name, pre_daemon_status, pre_daemon_pid))
+
+    duthost.stop_pmon_daemon(daemon_name, SIG_STOP_SERVICE)
+    time.sleep(2)
+
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    pytest_assert(daemon_status == expected_stopped_status,
+                          "{} expected stopped status is {} but is {}".format(daemon_name, expected_stopped_status, daemon_status))
+    pytest_assert(daemon_pid == -1,
+                          "{} expected pid is -1 but is {}".format(daemon_name, daemon_pid))
+
+    data = collect_data(duthost)
+    pytest_assert(not data['keys'], "DB data keys is not cleared on daemon stop")
+    pytest_assert(not data['data'], "DB data is not cleared on daemon stop")
+
+    duthost.start_pmon_daemon(daemon_name)
+    time.sleep(10)
+
+    post_daemon_status, post_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    pytest_assert(post_daemon_status == expected_running_status,
+                          "{} expected restarted status is {} but is {}".format(daemon_name, expected_running_status, post_daemon_status))
+    pytest_assert(post_daemon_pid != -1,
+                          "{} expected pid is -1 but is {}".format(daemon_name, post_daemon_pid))
+    pytest_assert(post_daemon_pid > pre_daemon_pid,
```

Comment:
Maybe post_daemon_pid != pre_daemon_pid?

Reply:
I am expecting that the PMON doesn't gets restarted because of this single daemon restarting. If the PMON docker doesn't get restarted then the pid will increase.